third_party/lua/luadebug.lua

```diff
@@ -538,6 +743,9 @@ local function cmd_quit()
 end
 
 local commands_help = {
+    {'b.reak.point $location|add_break.point', 'set new breakpoints at module.lua+num', cmd_add_breakpoint},
+    {'bd.elete $location|delete_break.point', 'delete breakpoints',  cmd_remove_breakpoint},
+    {'bl.ist|list_break.points', 'list breakpoints', cmd_list_breakpoints},
```

Comment:
This is not patch-specific, but from the usability perspective, the current help listing is quite bad:
```
luadebug.lua> help
b|break|breakpoint|add_break|add_breakpoint $location
    - set new breakpoints at module.lua+num
bd|bdelete|delete_break|delete_breakpoint $location
    - delete breakpoints
bl|blist|list_break|list_breakpoints 
    - list breakpoints
c|cont|continue 
    - continue execution
d|down 
    - move down the stack by one frame
e|eval $expression
    - execute the statement
f|finish|step_out 
    - step forward until exiting the current function
h|help|? 
    - print this help message
l|locals 
    - print the function arguments, locals and upvalues
n|next|step_over 
    - step forward by one line (skipping over functions)
p|print $expression
    - execute the expression and print the result
q|quit 
    - exit debugger
s|st|step|step_into 
    - step forward by one line (into functions)
t|trace|bt 
    - print the stack trace
u|up 
    - move up the stack by one frame
w|where $linecount
    - print source code around the current line
```

It is hard for me to read this, `gdb` on the other hand outputs commands and descriptions on the same line:
```
List of commands:

awatch -- Set an access watchpoint for EXPRESSION.
break, brea, bre, br, b -- Set breakpoint at specified location.
break-range -- Set a breakpoint for an address range.
catch -- Set catchpoints to catch events.
catch assert -- Catch failed Ada assertions, when raised.
catch catch -- Catch an exception, when caught.
catch exception -- Catch Ada exceptions, when raised.
catch exec -- Catch calls to exec.
catch fork -- Catch calls to fork.
catch handlers -- Catch Ada exceptions, when handled.
catch load -- Catch loads of shared libraries.
catch rethrow -- Catch an exception, when rethrown.
catch signal -- Catch signals by their names and/or numbers.
catch syscall -- Catch system calls by their names, groups and/or numbers.
```

Personally, I would even align the descriptions, so user gets a table-like info.

Reply:
Table like output, with all aliases mentioned, and with description displayed on the same line, would not fit to the window. So we had to place command description to the next line.

OTOH, I agreed that using '|' as delimiter without any extra space is not very easy to read, even if they will be displayed using different colors. So I've made them looking the following way:

![image](https://user-images.githubusercontent.com/4956130/214699960-7460e555-7283-4759-959c-ed6d0a9f45f2.png)

Please see spaces around delimiters, and different colors for command, aliases and description.

Reply:
```diff
diff --git a/third_party/lua/luadebug.lua b/third_party/lua/luadebug.lua
index ae8ab54ea..3cc444b69 100644
--- a/third_party/lua/luadebug.lua
+++ b/third_party/lua/luadebug.lua
@@ -725,15 +725,17 @@ local function cmd_help()
         if #map.aliases > 0 then
             local fun = require 'fun'
             local txt = ''
-            fun.each(function(x) txt = txt .. '|' .. color_yellow(x) end,
+            fun.each(function(x) txt = txt .. ', ' .. color_yellow(x) end,
                      map.aliases)
-            print(color_blue(v) .. '|' .. string.sub(txt, 2, #txt) .. ' ' ..
-                  (map.arg or ''))
+            dbg.writeln(color_blue(v) .. ', ' .. string.sub(txt, 3, #txt) ..
+                        ' ' .. (map.arg or ''))
         else
-            print(color_blue(v) .. ' ' .. (map.arg or ''));
+            dbg.writeln(color_blue(v) .. ' ' .. (map.arg or ''));
         end
-        print('    - ' .. map.help)
+        dbg.writeln(color_grey('    -- ' .. map.help))
     end
+    dbg.writeln('')
+
     return false
 end
```

Reply:
Well, the lines seem to be too long because of excessive names. Are you sure we need to list all long aliases?
```
b, break, breakpoint, add_break, add_breakpoint $location
```